tools/Vrops.py

```diff
@@ -149,368 +124,120 @@ def get_resources(self, target, token, uuids, resourcekinds):
         return self.get_resources(target, token, parent_uuids, resourcekinds=["Datacenter"])
 
-    def get_vccluster(self, target, token, parent_uuids):
-        return self.get_resources(target, token, parent_uuids, resourcekinds=["ClusterComputeResource"])
+    def get_cluster_and_datastores(self, target, token, parent_uuids):
+        return self.get_resources(target, token, parent_uuids, resourcekinds=["ClusterComputeResource", "Datastore"])
 
     def get_hosts(self, target, token, parent_uuids):
         return self.get_resources(target, token, parent_uuids, resourcekinds=["HostSystem"])
 
-    def get_vms_and_ds(self, target, token, parent_uuids):
-        return self.get_resources(target, token, parent_uuids, resourcekinds=["Datastore", "VirtualMachine"])
+    def get_vms(self, target, token, parent_uuids):
+        return self.get_resources(target, token, parent_uuids, resourcekinds=["VirtualMachine"])
 
-    def get_project_ids(target, token, uuids, collector):
-        logger.debug('>---------------------------------- get_project_ids')
-        logger.debug(f'target   : {target}')
-        logger.debug(f'collector: {collector}')
+    def get_latest_values_multiple(self, target: str, token: str, uuids: list, keys: list, collector: str,
+                                   kind: str = None) -> (list, int):
+
+        # vrops can not handle more than 1000 uuids for stats
+        uuids_chunked = list(chunk_list(uuids, 1000)) if kind == 'stats' else [uuids]
+
+        url = f"https://{target}/suite-api/api/resources/stats/latest/query" if kind == 'stats' else \
+            f"https://{target}/suite-api/api/resources/properties/latest/query"
 
-        if not isinstance(uuids, list):
-            logger.error('Error in get project_ids: uuids must be a list with multiple entries')
-            return False
-        # vrops can not handle more than 1000 uuids
-        uuids_chunked = list(chunk_list(uuids, 1000))
-        project_ids = list()
-        url = "https://" + target + "/suite-api/api/resources/bulk/relationships"
         headers = {
             'Content-Type': "application/json",
             'Accept': "application/json",
-            'Authorization': "vRealizeOpsToken " + token
+            'Authorization': f"vRealizeOpsToken {token}"
         }
-        import queue
+
         q = queue.Queue()
         thread_list = list()
         chunk_iteration = 0
 
+        logger.debug('>----------------------- get_latest_values_multiple')
+        logger.debug(f'target   : {target}')
+        logger.debug(f'collector: {collector}')
+        logger.debug(f'Amount keys : {len(keys)}')
+        for k in keys:
+            logger.debug(f'key  : {k}')
+
         for uuid_list in uuids_chunked:
             chunk_iteration += 1
-            t = Thread(target=Vrops.get_project_id_chunk,
-                       args=(q, uuid_list, url, headers, target, chunk_iteration))
+            t = Thread(target=Vrops._get_chunk,
```

Comment:
Am I getting this right, that `uuid_list` would be only one in case of properties? Hence no chunking

Reply:
Yes, thats right. We are able to fetch 10000 resources at once and it is working. Stats need to be chunks. Don't know why.